pkg/cloud/services/ec2/instances.go

```diff
@@ -420,13 +420,16 @@ func (s *Service) GetCoreNodeSecurityGroups(scope *scope.MachinePoolScope) ([]st
 	}
 
 	ids := make([]string, 0, len(sgRoles))
-	for _, sg := range sgRoles {
-		if _, ok := s.scope.SecurityGroups()[sg]; !ok {
-			return nil, awserrors.NewFailedDependency(
-				fmt.Sprintf("%s security group not available", sg),
-			)
+	existingSGs := s.scope.SecurityGroups()
```

Comment:
I'm curious as to why this part has been changed as part of this?

Reply:
There are 2 reasons for this
1. In old code we are accessing `s.scope.SecurityGroups()[sg]` where `s.scope.SecurityGroups()` is a slice. And as in former condition we are checking length of `s.scope.SecurityGroups()` so it might given index out of range exception if `len(s.scope.SecurityGroups()) == 0`
2. And as we are calling `s.scope.SecurityGroups()` multiple times so thought to store it in a variable `exisitingSGs`

Reply:
> In old code we are iterating over sgRoles slice and then check if that role exisit in existingSGs or not. But in case of len(existingSGs)==0 then there is no need to range over sgRoles slice.

If the `sgRoles` length is 0, then automatically it wont range over `sgroles` and skip this block, so I think we dont need to do explicit check for length.

Reply:
Thanks everyone for suggestion. Now it makes sense to me as `s.scope.SecurityGroups()` is only accessing infra cluster spec, so no issues in calling it multiple times and also no use of checking length for existingSGs as when there is no element in the map then accessing its element return ok=false.